Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift

```diff
@@ -43,12 +58,55 @@ class FetchSnapcodeViewController: UIViewController, UITextFieldDelegate {
         usernameTextField.delegate = self
         view.addSubview(usernameTextField)
         
-        nextButton = UIButton(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: 50))
-        nextButton.setTitle("Next", for: .normal)
-        nextButton.setTitleColor(UIColor.snapBlack, for: .normal)
-        nextButton.setBackgroundImage(UIImage(color: UIColor.snapYellow, size: nextButton.frame.size), for: .normal)
-        nextButton.setBackgroundImage(UIImage(color: UIColor.snapYellow.withAlphaComponent(0.75), size: nextButton.frame.size), for: .highlighted)
-        nextButton.addTarget(self, action: #selector(nextButtonWasPressed), for: .touchUpInside)
+        getSnapcodeButton = UIButton(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: 50))
+        getSnapcodeButton.setTitle("Get Snapcode", for: .normal)
+        getSnapcodeButton.setTitleColor(UIColor.snapBlack, for: .normal)
+        getSnapcodeButton.setBackgroundImage(UIImage(color: UIColor.snapYellow, size: getSnapcodeButton.frame.size), for: .normal)
+        getSnapcodeButton.setBackgroundImage(UIImage(color: UIColor.snapYellow.withAlphaComponent(0.75), size: getSnapcodeButton.frame.size), for: .highlighted)
+        getSnapcodeButton.addTarget(self, action: #selector(getSnapcodeButtonWasPressed), for: .touchUpInside)
+        
+        fetchSnapcodeIndicator = JGProgressHUD(style: .dark)
+        fetchSnapcodeIndicator.contentView.backgroundColor = UIColor.snapBlack
+        fetchSnapcodeIndicator.textLabel.text = "Fetching Snapcode"
+        
+        //Snapcode Selection UI
+        snapcodeImageView = UIImageView()
+        snapcodeImageView.frame = CGRect(x: 0.0, y: System.shared.statusBarHeight + usernameTextField.frame.height + 10.0, width: view.frame.width * 0.75, height: view.frame.width * 0.75)
+        snapcodeImageView.center.x = view.center.x
+        
+        snapcodeSelectionToolbar = UIToolbar(frame: CGRect(x: 0, y: view.frame.height - 45, width: view.frame.size.width, height: 45))
+        snapcodeSelectionToolbar.isTranslucent = false
+        snapcodeSelectionToolbar.barTintColor = UIColor.snapBlack
+        
+        let cancelBarButton = UIBarButtonItem(title: "Back", style:.plain, target: self, action: #selector(cancelButtonWasPressed))
+        cancelBarButton.tintColor = UIColor.snapYellow
+        
+        let nextButton: UIBarButtonItem = UIBarButtonItem(title: "Next", style:.plain, target: self, action: #selector(nextButtonWasPressed))
+        nextButton.tintColor = UIColor.snapYellow
+        
+        let flexibleSpaceBarButtonItem = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: self, action: nil)
```

Comment:
we use this enough times that we should make this an extension of UIBarButtonItem. You should be able to call it with UIBarButtonItem.flexibleSpace() and it would initialize with target:nil and action:nil